### PR TITLE
ListK docs

### PR DESCRIPTION
**Pull request by pakoito**
It's a glaring omission and at this point we just *need* it.

### CODE REVIEW

**Comment by pakoito** [comment]
Closes #322

File: modules/docs/arrow-docs/docs/docs/datatypes/listk/README.md [hunk 1]
```diff
@@ -6,4 +6,89 @@ permalink: /docs/datatypes/listk/
 
 ## ListK
 
-TODO. Meanwhile you can find a short description in the [intro to datatypes]({{ '/docs/datatypes/intro/' | relative_url }}).
+ListK wraps over the platform `List` type to make it a (type constructor)[/docs/patterns/glossary/#type-constructors].
+
+It can be created from Kotlin Sequence type with a convenient `k()` function.
```
Comment (nomisRev) [inline modules/docs/arrow-docs/docs/docs/datatypes/listk/README.md]: Sequence > List

File: modules/docs/arrow-docs/docs/docs/datatypes/listk/README.md [hunk 2]
```diff
@@ -6,4 +6,89 @@ permalink: /docs/datatypes/listk/
...
+hello.combineK(commaSpace.combineK(world)) == hello.combineK(commaSpace).combineK(world)
+```
+
+The functions `traverse` and `sequence` come from [`Traverse`](/docs/typeclasses/traverse/).
+
+Traversing a list creates a new container (`Kind<F, A>`)[/docs/patterns/glossary/#type-constructors] by combining the result of a function applied to each element:
+
+```kotlin:ank
+val numbers =  listOf(Math.random(), Math.random(), Math.random()).k()
+requests.traverse(Option.applicative()) { if (it > 0.5) Some(it) else None }
```
Comment (nomisRev) [inline modules/docs/arrow-docs/docs/docs/datatypes/listk/README.md]: `numbers.traverse`?

File: modules/docs/arrow-docs/docs/docs/datatypes/listk/README.md [hunk 3]
```diff
@@ -6,4 +6,89 @@ permalink: /docs/datatypes/listk/
...
+```
+
+and complements the convenient function `sequence()` that converts a list of `List<Kind<F, A>>` into a `Kind<F, List<A>>`:
```
Comment (nomisRev) [inline modules/docs/arrow-docs/docs/docs/datatypes/listk/README.md]: ListK<Kind<F, A>> into a Kind<F, ListK<A>>?

File: modules/docs/arrow-docs/docs/docs/datatypes/listk/README.md [hunk 4]
```diff
@@ -6,4 +6,89 @@ permalink: /docs/datatypes/listk/
...
+
+```kotlin:ank
+fun request() = Some(Math.random())
+
+val requests =  listOf(request(), request(), request()).k()
+requests.sequence(Option.applicative()).fix()
+```
+
+A more general approach is aggregating values using `foldLeft` and `foldRight` from [`Foldable`](/docs/typeclasses/foldable).
```
Comment (nomisRev) [inline modules/docs/arrow-docs/docs/docs/datatypes/listk/README.md]: Not sure I understand this sentence right. Should it be?

> A more general approach **for** aggregating values?

Comment (pakoito) [reply modules/docs/arrow-docs/docs/docs/datatypes/listk/README.md]: I'll rewrite.

**Pull request merged**